### PR TITLE
feat: Add recent search query

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Save recent search location by Preference
- Improve UI in SearchLocationFragment
- I set up RecyclerView instead of ListView as I found problem allowing nestedScrollView enabled with ListView, which made ListView only showing the first item

Fixes: #575

Screenshots for the change:
ezgif-2-22edb31efa62

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 1]
```diff
@@ -187,14 +232,51 @@ class SearchLocationFragment : Fragment() {
...
+                rootView.popularLocationsLayout.isVisible = list.isNotEmpty()
             })
 
-        rootView.eventLocationLv.setOnItemClickListener { parent, view, position, id ->
-            savePlaceAndRedirectToMain(eventLocationList[position])
+        searchLocationViewModel.loadEventsLocation()
+    }
+}
+
+class LocationsAdapter : RecyclerView.Adapter<LocationViewHolder>() {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: IMO it's better to create a new file for a class.

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @liveHarshit updated

File: app/src/main/res/layout/fragment_search_location.xml [hunk 2]
```diff
@@ -1,141 +1,184 @@
...
+                    android:textStyle="bold"
+                    android:textSize="@dimen/text_size_expanded_title" />
+
+                <androidx.recyclerview.widget.RecyclerView
+                    android:id="@+id/popularLocationsRv"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:dividerHeight="0dp"
+                    android:divider="@null"
+                    android:layout_marginBottom="8dp" />
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_search_location.xml]: In dimensions resources file

Comment (anhanh11001) [reply app/src/main/res/layout/fragment_search_location.xml]: fixed

**Pull request merged**